mlperf_logging/rcp_checker/rcp_checker.py

```diff
@@ -0,0 +1,339 @@
+                              record_contents['Benchmark'],
+                              record_contents['RCP Mean'],
+                              record_contents['RCP Stdev'],
+                              len(epoch_list)-2)
+            record_contents['Max Speedup'] = record_contents['RCP Mean'] / min_epochs
+            #print(record, record_contents)
+            #print("\n")
+
+    def _find_rcp(self, benchmark, bs):
+        '''Find RCP based on benchmark and batch size'''
+        for _, record_contents in self.rcp_data.items():
+            if record_contents['Benchmark'] == benchmark and record_contents['BS'] == bs:
+                return record_contents
+
+
+    def _find_min_rcp(self, benchmark):
+        '''Find RCP with the smallest batch size for a benchmark'''
+        min_bs = 1e9
+        min_record = None
+        for _, record_contents in self.rcp_data.items():
+            if record_contents['BS'] < min_bs and record_contents['Benchmark'] == benchmark:
+                min_record = record_contents
+                min_bs = record_contents['BS']
+        return min_record
+
+    def _find_top_min_rcp(self, benchmark, bs):
+        '''
+        Find top RCP to serve as min in interpolation.
+        For example, if bs = 100 and reference has bs = 10, 20, 110, 120
+        this will return the RCP with bs = 20.
+        '''
+        min_bs = 0
+        min_record = None
+        for _, record_contents in self.rcp_data.items():
+            if record_contents['Benchmark'] == benchmark:
+                if record_contents['BS'] < bs and record_contents['BS'] > min_bs:
+                    min_bs = record_contents['BS']
+                    min_record = record_contents
+        return min_record
+
+
+    def _find_bottom_max_rcp(self, benchmark, bs):
+        '''
+        Find bottom RCP to serve as max in interpolation.
+        For example, if bs = 100 and reference has bs = 10, 20, 110, 120
+        this will return the RCP with bs = 110.
+        '''
+        max_bs = 1e9
+        max_record = None
+        for _, record_contents in self.rcp_data.items():
+            if record_contents['Benchmark'] == benchmark:
+                if record_contents['BS'] > bs and record_contents['BS'] < max_bs:
+                    max_bs = record_contents['BS']
+                    max_record = record_contents
+        return max_record
+
+
+    def _find_p_value(self, subm_mean, subm_stdev, subm_num_samples,
+                      ref_mean, ref_stdev, ref_num_samples,
+                      p_value_lim=0.05):
+        '''
+        Do t-test between submission and reference and return p-value and
+        whether it is larger than the limit
+        https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind_from_stats.html
+        '''
+        t_stat, p_value = scipy.stats.ttest_ind_from_stats(
+                              subm_mean, subm_stdev, subm_num_samples,
```

Comment:
This is a specific point of contention, wether we use "subm_stdev" or use "ref_stdev" twice. 

Both Peter and I have both advocated for using "ref_stdev" in place of "subm_stdev" as counter-intuitive as that seems - because it means that the submitter just needs to reason about their mean and we can do back-of-the-envelope calculations easier. 

This is slight perversion of good statistics and science, though. 

I'd recommend making this change and sending a note to John to check for objection on this specific point.

Reply:
Also - double check with Peter to make sure he's not being a moving target here.

Reply:
I am quite confused here. In practice we do use ref_stdev twice, because this function is called to find the max speedup for any submission, so it is not aware of any submission. See line 151:            
 _, p_value = self._find_p_value(
                             mean_mid, stdev, num_samples_ref,
                             mean, stdev, num_samples_subm,
                             0.05)
What I am not sure about is whether I should use num_samples_subm, or use twice num_samples_ref (we know the population sizes). Let me know what you think

Reply:
I talked to Peter about it, the best is to have an 1-off meeting to figure this out.